### PR TITLE
Setup

### PR DESCRIPTION
**Pull request by Alejandro-Bernal-M**
# Database based on the diagram.

For this exercise, I implemented a database based on the next diagram:
diagram

with the following results: 
postgres_result

## Working solo

I have the authorization for working alone because my partner had some personal problems.
image

### CODE REVIEW

File: BasedOnDiagram.sql [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    FOREIGN KEY ("patient_id")
+      REFERENCES "patients"("id")
+);
+
+CREATE TABLE "invoices" (
+  "id" int,
+  "total_amount" decimal,
+  "generated_at" timestamp,
+  "payed_at" timestamp,
+  "medical_history_id" int,
```
Comment (Zuhaib042) [inline BasedOnDiagram.sql:24]: - [ ] This is a foreign key so kindly consider setting it up as a foreign key. it references medical_histories(id). consider fixing it.

File: BasedOnDiagram.sql [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+CREATE TABLE "medical_histories" (
+  "id" int,
+  "admitted_at" timestamp,
+  "patient_id" int,
+  "status" varchar(40),
+  PRIMARY KEY ("id"),
+  CONSTRAINT "FK_medical_histories.patient_id"
+    FOREIGN KEY ("patient_id")
+      REFERENCES "patients"("id")
```
Comment (Zuhaib042) [inline BasedOnDiagram.sql:16]: - [ ] Good job on adding foreign keys, but I dont see indexes for any foreign key. kindly consider adding indexes for all the foreign keys in their respective table. I am requiring this not just here but for all the foreign keys.
for example : CREATE INDEX ON medical_histories (patient_id);
requirement:
Remember to add the FK indexes.

**Review by Zuhaib042 — CHANGES_REQUESTED** [review]
Hi @Alejandro-Bernal-M 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✅ Implemented most of the requirements
✅ Professional PR and Readme

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Shedrack-Sunday — APPROVED** [review]
Hi @Alejandro-Bernal-M 

Congratulations! 🎉 Your project is complete and ready to be merged! 🚀

image


## Highlights
- Good use of SQL queries ✔️ 
- Proper use of GitHub PR  features ✔️ 


## Optional Suggestions 💡
While these comments with the [OPTIONAL] prefix won't prevent the approval of this pull request, I strongly recommend taking them into account as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really enhance your application.
Cheers, and Happy coding!👏👏👏

If you have any questions or need clarification on anything, feel free to leave a comment in the pull request thread, and don't forget to tag me so I'll receive the notification.

As per the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project. If you feel that the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**